src/collection.cr

```diff
@@ -240,8 +240,34 @@ struct YAML::Any
   end
 end
 
+module Digable
+  def dig(key : K, *subkeys)
+    if (value = self[key]) && value.responds_to?(:dig)
+      return value.dig(*subkeys)
+    end
+    raise KeyError.new "#{self.class} value not diggable for key: #{key.inspect}"
+  end
+
+  # :nodoc:
+  def dig(key : K)
+    self[key]
+  end
+
+  def dig?(key : K, *subkeys)
+    if (value = self[key]?) && value.responds_to?(:dig?)
+      value.dig?(*subkeys)
+    end
+  end
+
+  # :nodoc:
+  def dig?(key : K)
+    self[key]?
+  end
+end
+
 class HTTP::Cookies
   include Collection(String, HTTP::Cookie)
+  include Digable
```

Comment:
What about popping this into an `included` macro in `Collection`, then including if the dig methods are not present?